internal/worker/Scheduler.go

```diff
@@ -0,0 +1,35 @@
+package worker
+
+import (
+	"context"
+	"log/slog"
+	"time"
+)
+
+func Scheduler(ctx context.Context, interval time.Duration, process func(ctx context.Context) error) {
+	ticker := time.NewTicker(interval)
+	defer ticker.Stop()
+
+	if ctx.Err() == nil {
+		err := process(ctx)
+		if err != nil {
+			slog.Error("Error during initial process execution", "error", err)
+		}
+	} else {
+		slog.Info("Stopping scheduler before first execution due to canceled context")
```

Comment:
please get the logger from the context and keep the log message in lowercase

```go
log := logger.GetLogger(ctx)
```

Reply:
Is it really necessary to display this log, what defence does it give the user?  Maybe turn this into an debug log?